src/core/public/notifications/toasts/toasts_api.tsx

```diff
@@ -143,6 +141,15 @@ export class ToastsApi implements IToasts {
    * @returns a {@link Toast}
    */
   public add(toastOrTitle: ToastInput) {
+    if (typeof toastOrTitle !== 'string') {
+      const list = this.toasts$.getValue();
+      const existingToast = list.find((toast) => toast.id === toastOrTitle.id);
+
+      if (existingToast) {
+        return existingToast;
+      }
+    }
+
     const toast: Toast = {
       id: String(this.idCounter++),
```

Comment:
I wonder if it's somewhat surprising that we still increment the `idCounter`, but throw away the value. On the one hand, it means that the counter really does match the total number, but on the other, there are now some idCounter values that don't exist in the map of toast IDs.

Reply:
We dont still increment, if an existing toast is found, we exit this function when we return the existing toast.

Reply:
Right, I'm talking about the case where an `id` is provided, but there's no existing matching toast. In that case we increment, but don't use the incremented value.